### PR TITLE
feat(cli): add support for --edit and pre-commit

### PR DESCRIPTION
**Pull request by cylewitruk**
I wanted to use this together with [pre-commit](https://pre-commit.com/) but noticed that support for `--edit` was missing, so I went ahead and added at least the default using `./.git/COMMIT_EDITMSG`.

Also added a `.pre-commit-hooks.yaml` so that the repo can be used together with `pre-commit`.  Nothing fancy here...

And, thanks for this! Much better than the node commitlint solution for rust projects :)

### CODE REVIEW

File: src/args.rs [hunk 1]
```diff
@@ -51,6 +51,15 @@ impl Args {
 
     /// Read commit messages from stdin.
     pub fn read(&self) -> Result<Vec<Message>, Error> {
+        // Check first whether or not the --edit option was supplied. When running from tooling such as
+        // `pre-commit`, stdin exists, so this needs to come first.
+        if self.edit {
+            let msg = std::fs::read_to_string("./.git/COMMIT_EDITMSG")
+                .expect("Failed to read './.git/COMMIT_EDITMSG");
```
Comment (KeisukeYamashita) [inline src/args.rs:58]:
```suggestion
                .expect("Failed to read './.git/COMMIT_EDITMSG'");
```

**Review by KeisukeYamashita — CHANGES_REQUESTED** [review]
Thank you for your PR!
Overall LGTM, I found a typo 🙏 

I'm new to `pre-commit`, can you kindly give me any document or something about this `--edit` flag, please 🙏

**Review by KeisukeYamashita — APPROVED** [review]
Thank you for your update!

**Comment by cylewitruk** [comment]
> Thank you for your update!

No problem :)

Once you create a new release (v.0.1.3), here's an example `.pre-commit-config.yaml` which can be used with `pre-commit` in case you wanted to add it to docs :)

```yaml
default_install_hook_types: [pre-commit, commit-msg, prepare-commit-msg]
repos:
- repo: https://github.com/KeisukeYamashita/commitlint-rs
  rev: v0.1.3
  hooks:
  - id: commitlint

```

**Comment by KeisukeYamashita** [comment]
@cylewitruk Sorry for the late reply 🙏 
Thanks, I'll update the doc 😄

**Pull request merged**